funding.json to apply to FLOSS/fund

shreekumar3d opened the pull request:
I've put together an initial version of funding.json to apply to FLOSS/fund.  Entity and project descriptions are based on PP website and v5 campaign pages.  I have extracted the financials from the report in Dutch (google translate a bit). Please reviewed and correct as appropriate.  I am putting things down a bit more in detail below.  

I have a local instance of FLOSS/fund software. I'm using it to simulate how Precious Plastic would look like on https://dir.floss.fund/, if we were to submit this json file.  All the text comes from funding.json.  For a simple comparison, lookup [blender on FLOSS/fund](https://dir.floss.fund/view/@www.blender.org). 

## Entity

Entity is the "organization" that owns projects.  This provides high level information about the Foundation and its goals. Contact info and web-page is on the right.

<img width="3024" height="857" alt="image" src="https://github.com/user-attachments/assets/a3f649a6-a9d8-41e8-9089-cd972a8c7c18" />

## Project

If you click on Projects, you will see that the Foundation is asking for funding for 1 project - which is PP v5. Only a short description of the project is visible here:

<img width="3024" height="857" alt="image" src="https://github.com/user-attachments/assets/341c86c5-e49c-488a-bcb1-1a41129ca7f5" />

If you click on the v5 project, you will see the details of the project, the license and tags:

<img width="3006" height="1412" alt="image" src="https://github.com/user-attachments/assets/38470d84-0b02-43e2-810d-f0b596753e69" />

## Funding Plans

If you click on the Funding Plans, you will see details on how folks can fund the project.  I have included Donorbox and Patreon for the "One time" and "Monthly" funding plans.  Bank is mentioned as the only funding channel for the large funders.   Clicking on the "Visit" links will take you to the Donorbox and the Patreon pages, as expected. 

<img width="2415" height="1960" alt="image" src="https://github.com/user-attachments/assets/8197423a-b8dd-4298-938e-005444739839" />

## History

Finally, the "History" tab gives everyone an idea of the finances of the project over time.

<img width="2968" height="1102" alt="image" src="https://github.com/user-attachments/assets/12ef2823-fdc7-480c-a82e-1241f8d3c2aa" />

## Review Required

These below items certainly need to be reviewed:

1. All the text and numbers - related to the foundation, project details, funding plans and descriptions. Tags for the project as well.
2.  I hacked up the numbers in the History tab from the [Dutch report](https://drive.google.com/drive/folders/1iQNo77MkYN_2v0FX3IVrp52a0lzHvrOV?usp=drive_link), using google translate!  The numbers certainly need to be checked for accuracy and corrected.
3. Do we have financial numbers from the previous years ? It would be nice to include them if possible.  This is one of the under-appreciated aspects of the FLOSS/fund.  Providing this data can help analysts direct information about what it took to build all these public goods.  PP being a decade old project - if we can share more data here, then that would be great !

## Miscellaneous

Finally, note the red exclamatory marks in the images.  We can fix that easily by hosting a text file at PP website under https://www.preciousplastic.com/.well-known/funding-manifest-urls . This file would need to have the below content:

`
https://github.com/ONEARMY/precious-plastic-kit/blob/master/funding.json
`

Essentially that serves to "prove" that the funding.json in the github repository indeed belongs to PP.  This URL will also need to be mentioned in the funding.json file under the "wellKnown" tag.  We'll come to that after fixing the text and numbers.  So please don't merge the PR just yet.

shreekumar3d committed:
floss/fund funding.json to signal funding needs

Entity and project descriptions are based on PP website and v5
campaign pages.  Financials are extracted from the report in Dutch.
Needs to be reviewed/corrected as appropriate.

shreekumar3d commented:
Regarding funding plans :  I coined the names "Boost" and "Mega Boost". Please check if we really want to list "Mega Boost" - 2 million here - that would technically make PP the project asking for the highest funding in floss/fund (see [table toppers in this link](https://floss-fund-live-status.streamlit.app/)) 

There's a big gap in funding plans between 10 EURO and 100k EURO.  In Donorbox, PP has 10,20,50,100,500 and 1000 EURO.  Would it make sense to port one of these as a funding plan and give it a name ? I can't think of any catchy names though.

sigolenej commented on funding.json:
```diff
@@ -0,0 +1,104 @@
+            "expenses": 53646,
+            "taxes": 0,
+            "currency": "EUR",
+            "description": "Financials for 2024"
```
Here are the adjust financial results for 2024 as well as for the previous years

 "year": 2024,
            "income": 214037,
            "expenses": 112129,
            "taxes": 0,
            "currency": "EUR",
            "description": "Financials for 2024"
 "year": 2023,
            "income": 240024,
            "expenses": 225986,
            "taxes": 0,
            "currency": "EUR",
            "description": "Financials for 2023"
"year": 2022,
            "income": 216500,
            "expenses": 251312,
            "taxes": 0,
            "currency": "EUR",
            "description": "Financials for 2022"

davehakkens commented:
thanks for the PR Shree!
Sigo left some comments. If you can intergrate those we can merge it!

shreekumar3d committed:
Updated with correct financials for 2022-24

shreekumar3d commented:
Thanks ! I've updated the financials in the json. We're almost there.  Need to host a file on your website at the following path:

https://www.preciousplastic.com/.well-known/funding-manifest-urls

The content of this file shall be (one line)

`https://github.com/ONEARMY/precious-plastic-kit/blob/master/funding.json`

So running wget https://www.preciousplastic.com/.well-known/funding-manifest-urls must return that content.

This serves to prove that preciousplastic.com is indeed associated with this github repo

(Noting this for context btw - I've also mentioned this in email)

davehakkens commented:
Thanks @shreekumar3d 
@sigolenej are we able to add that file to Webflow just as we did recently for Apply Pay?

sigolenej commented:
Sure @shreekumar3d could you share the file you need to be uploaded with the desired name, format, etc.? 
Thanks !

sigolenej commented:
Hey @shreekumar3d checking in if you saw my previous comment

shreekumar3d commented:
@sigolenej was travelling. Just returned yesterday after a break. To elaborate on what's needed,  it's a text file with a single line of content (same as the URL of the funding.json file after this PR merges)

`https://github.com/ONEARMY/precious-plastic-kit/blob/master/funding.json`

The file must be hosted on the following URL on your webserver

https://www.preciousplastic.com/.well-known/funding-manifest-urls

As a simple example, you may refer to what the OpenSSL project did for their funding.json.  Their well-known known is hosted at https://openssl-foundation.org/.well-known/funding-manifest-urls . So, when I run
```
$ wget https://openssl-foundation.org/.well-known/funding-manifest-urls 
....
  HTTP/1.1 200 OK
  content-type: text/plain
....
```

I get

```
$ cat https://openssl-foundation.org/.well-known/funding-manifest-urls
https://github.com/openssl/openssl/blob/master/funding.json
```

shreekumar3d commented:
[This link](https://fundingjson.org/#:~:text=main/funding.json-,wellKnown,-If%20a) has some documentation about well-known, just for reference.

sigolenej commented:
@shreekumar3d please confirm the file has been correctly uploaded :)

shreekumar3d committed:
Add wellknown manifest reference

This basically "proves" that we own those URLs. For this to work, e.g.

wget https://www.preciousplastic.com/.well-known/funding-manifest-urls

must return a single line where the github repo is hosted:

https://github.com/ONEARMY/precious-plastic-kit/blob/master/funding.json

This is required as the domains don't match.

shreekumar3d commented:
@sigolenej  - yes, I confirm this works. Thank you!

@davehakkens - the PR may now be merged... all the changes are now done!
